django/db/backends/mysql/creation.py

```diff
@@ -48,7 +41,11 @@ def _clone_test_db(self, suffix, verbosity, keepdb=False):
         with self._nodb_connection.cursor() as cursor:
             try:
                 self._execute_create_test_db(cursor, test_db_params, keepdb)
-            except Exception:
+            except Exception as e:
+                if not self._is_db_exists_error(e):
+                    raise
```

Comment:
I don't think we need these two lines:
```python
if not self._is_db_exists_error(e):
     raise
```
because if an exception other than `"database exists" (1007)` is raised in `super()._execute_create_test_db()`, then tests will be stopped earlier by `sys.exit(2)`.

Reply:
I'm not sure if it's important but what if these lines
https://github.com/django/django/blob/b4c8dc279a85b9bed35c5679f75b1c4c3432b537/django/db/backends/mysql/creation.py#L28-L31

raise some exception (not SystemExit)?

Reply:
I do not think it is important and I'm not sure if it is possible :thinking: